### PR TITLE
Servicefoundation fixes

### PR DESCRIPTION
**Pull request by Prutswonder**
* Updated the package manager and packages to their latest version
* Updated readme & makefile
* Added Histogram.RecordDuration for histograms with custom durations.
* Fixed a lock issue in CountLabels that for some reason panicked in the latest Go version.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,10 +1,31 @@
+.PHONY: restore run-tests cover vet lint
+
+COVERALLS_TOKEN := "DJljOZEYgGX0vkvoPCVNxyIupLJ6VscoO"
```
Comment (baor) [inline Makefile]: > For a public github repository, it is not necessary to define your repository key (COVERALLS_TOKEN).

https://github.com/mattn/goveralls

File: Metrics.go [hunk 2]
```diff
@@ -209,12 +219,26 @@ func (ctx *Metrics) addHistogramWithBuckets(subsystem, name, help string, bucket
...
 
+// RecordDuration adds the elapsed time since the specified start to the histogram in the specified unit of time
+// and to the linked summary in milliseconds.
+func (histogram *MetricsHistogram) RecordDuration(start time.Time, unit time.Duration) {
+	since := time.Since(start)
+	elapsedSeconds := float64(since.Seconds())
+	elapsedUnits := float64(since.Truncate(unit))
+
+	histogram.hist.Observe(elapsedUnits)
+	histogram.sum.Observe(elapsedSeconds * 1000.0)
```
Comment (baor) [inline Metrics.go]: Is looks like seconds are multiplied by 1000. Is it correct?

Comment (Prutswonder) [reply Metrics.go]: Unfortunately, yes. It's to convert seconds to milliseconds. Normally we would use a time constant like `time.Second`, but there's no constant for milliseconds.

File: Makefile (hunk 1 above)
Comment (Prutswonder) [reply Makefile]: You're right. I removed it.

**Pull request merged**